src/main/java/frc/robot/commands/DriveDistance.java

```diff
@@ -0,0 +1,71 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj.PIDController;
+import edu.wpi.first.wpilibj.PIDOutput;
+import edu.wpi.first.wpilibj.command.Command;
+import frc.robot.Robot;
+import frc.robot.TwoEncoderPIDSource;
+
+public class DriveDistance extends Command {
+  PIDController DriveDistanceController;
+  PIDOutput DriveDistanceOutput;
+   double distance, kP, tolerance;
+
+  public DriveDistance(double distance, double kP, double tolerance) {
+    requires(Robot.driveTrain);
```

Comment:
Why is the kp and tolerance in the constructor? Do they change for every instance?

Reply:
hint: the answer might be yes, but you'd have to convince me....